### DISCUSSION TITLE
[Frontend] Structuring Storybook

### DISCUSSION THREAD

**Discussion by rkskekzzz** [80000Coding/80000Coding-Web-Client#121] [Ideas]
# 현재 상태
파일 구별이 명확하지 않아서 새로운 스토리북을 생성할 때, 어떤 위치, 어떤 방식으로 생성해야할지 결정하는데 시간이 많이 걸립니다
명확하게 구조를 세워둘 필요가 있어 보입니다!

<img width="314" alt="image" src="https://github.com/80000Coding/80000Coding-Web-Client/assets/57925497/e218a15d-84b1-42ad-a1cb-68af6beea29f">

# 예시
[Storybook에서 공식문서로 구성하는 가이드를 작성해 두었습니다!](https://storybook.js.org/blog/structuring-your-storybook/)

아래는 위 링크에 나오는 모든 예시들입니다 :)

### Atomic Design methodology
- [Codecademy](https://gamut.codecademy.com/?ref=storybookblog.ghost.io)
- [The Guardian](https://5dfcbf3012392c0020e7140b-gmgigeoguh.chromatic.com/?path=%2Fstory%2Flayouts-showcase--article-story&ref=storybookblog.ghost.io)
- [United Nations World Food Programme](https://uikit.wfp.org/docs/index.html?path=%2Fstory%2Fgetting-started-intro--page&ref=storybookblog.ghost.io)

### By functionality
- [Spotify Backstage](https://backstage.io/storybook/?ref=storybookblog.ghost.io)
- [Grommet](https://storybook.grommet.io/?path=%2Fstory%2Fall--all&ref=storybookblog.ghost.io)
- [Monday.com Vibe](https://style.monday.com/?ref=storybookblog.ghost.io)

### By component status
- [IBM Carbon](https://react.carbondesignsystem.com/?path=%2Fstory%2Fgetting-started--welcome&ref=storybookblog.ghost.io)
- [Workday Canvas](https://workday.github.io/canvas-kit/?path=%2Fstory%2Fwelcome-getting-started--page&ref=storybookblog.ghost.io)

**Comment by rkskekzzz** [comment]
위 예시 중 마음에 드는 샘플을 골라서 해당 샘플의 구조를 따라하면 좋을 것 같습니다! 많은 의견 부탁드립니당:)

**Comment by leekh716** [comment]
오 저는 맨 처음 codeacademy꺼 맘에 드는데요!
컴포넌트 단계에 따라 atom -> molecule -> organism 로 구분한게 직관적이고 이해가 잘 가는 구성이네요(생명공학 전공이라 그런건 비밀)

**Reply by rkskekzzz** [reply]
저도 첫번째 방식이 젤 좋아보이네요!!
추가로 custom hook에 대한 가이드를 아래 링크처럼 만들어 두는 것도 좋아보입니다!
https://react.carbondesignsystem.com/?path=/docs/hooks-usecontextmenu--overview
